src/libraries/System.Private.CoreLib/src/System/Globalization/Normalization.cs

```diff
@@ -44,6 +44,11 @@ internal static bool TryNormalize(ReadOnlySpan<char> source, Span<char> destinat
         {
             CheckNormalizationForm(normalizationForm);
 
+            if (source.Overlaps(destination))
+            {
+                throw new ArgumentException(SR.Argument_OverlapSpansAreNotAllowed, nameof(destination));
```

Comment:
We have prior art for this: https://github.com/search?q=repo%3Adotnet%2Fruntime%20InvalidOperation_SpanOverlappedOperation&type=code

Do the same here?